Sources/NIOExtras/LengthFieldBasedFrameDecoder.swift

```diff
@@ -161,17 +166,27 @@ public final class LengthFieldBasedFrameDecoder: ByteToMessageDecoder {
     /// - parameters:
     ///    - buffer: The buffer containing the integer frame length.
     ///
-    private func readFrameLength(for buffer: inout ByteBuffer) -> Int? {
+    private func readFrameLength(for buffer: inout ByteBuffer) throws -> Int? {
 
         switch self.lengthFieldLength {
         case .one:
             return buffer.readInteger(endianness: self.lengthFieldEndianness, as: UInt8.self).map { Int($0) }
```

Comment:
oh, that's not your fault but there are quite a few bugs here. We can't make an `Int` out of all of these values without potentially crashing. We need to store the length as `UInt64` on all platforms or establish some maximum. I'd be happy if we say the max supported is `UInt32.max` = 4 GB on all platforms.

Reply:
We can't store the length in a `UInt64` because we can't pass that to `readSlice`. Having the Int fail to fit into UInt64 is fine: any slice that actually flips the top bit of a UInt64 is obviously ridiculous.

Reply:
@Lukasa I see! Should we just say the maximum supported is 2GB and define that as the universal maximum on all platforms? Then we can continue to use `Int` and just need to check (as done in this PR). But we should still support 64 length fields on 32 bit.

Reply:
SGTM.

Reply:
I have removed the precondition check in the initialiser. 64 bit length fields are now supported on 32-bit platforms.
Should we really add an additional check and throw an error if a 64 bit length field exceeds 2GB on a 64-bit platform?
`ByteToMessageHandler` in apple/swift-nio already has the functionality to limit the maximum supported size.

Reply:
@dnadoba I believe we should add this check for a couple of reasons:
- 32 and 64 bit APIs match and both support 2GB and no more
- the max we support in ByteBuffer is 4GB (BBs can't grow more than 4GB)

Reply:
I should expand on the second point: Let's say the user were to allow up to 5 GB here (on 64bit platforms), then ByteBuffer's `write*` would crash on the `write*` that crosses 4 GB anyway. A user will not know this information so I think we should catch this here.

Reply:
Got it, thanks for clarifying!